src/Tools/Source/RunTests/Cache/ContentUtil.cs

```diff
@@ -0,0 +1,143 @@
+            AppendReferences(builder, assemblyPath);
+            builder.AppendLine("Options:");
+            builder.AppendLine($"\t{nameof(_options.Test64)} - {_options.Test64}");
+            builder.AppendLine($"\t{nameof(_options.UseHtml)} - {_options.UseHtml}");
+            builder.AppendLine($"\t{nameof(_options.Trait)} - {_options.Trait}");
+            builder.AppendLine($"\t{nameof(_options.NoTrait)} - {_options.NoTrait}");
+
+            // TODO: Need to include dependency information here, option data, etc ...
+            // Test file alone isn't enough.  Makes it easy to test though.
+            return builder.ToString();
+        }
+
+        private void AppendReferences(StringBuilder builder, string assemblyPath)
+        {
+            builder.AppendLine("References:");
+
+            var initialAssembly = Assembly.ReflectionOnlyLoadFrom(assemblyPath);
+            var set = new HashSet<string>();
+            var toVisit = new Queue<AssemblyName>(initialAssembly.GetReferencedAssemblies());
+            var binariesPath = Path.GetDirectoryName(initialAssembly.Location);
+            var references = new List<Tuple<string, string>>();
+
+            while (toVisit.Count > 0)
+            {
+                var current = toVisit.Dequeue();
+                if (!set.Add(current.FullName))
+                {
+                    continue;
+                }
+
+                try
+                {
+                    var currentPath = Path.Combine(binariesPath, Path.ChangeExtension(current.Name, "dll"));
+                    var currentAssembly = File.Exists(currentPath)
+                        ? Assembly.ReflectionOnlyLoadFrom(currentPath)
+                        : Assembly.ReflectionOnlyLoad(current.FullName);
+
+                    foreach (var name in currentAssembly.GetReferencedAssemblies())
+                    {
+                        toVisit.Enqueue(name);
+                    }
+
+                    var currentHash = GetFileChecksum(currentAssembly.Location);
+                    references.Add(Tuple.Create(current.Name, currentHash));
+                }
+                catch
+                {
+                    references.Add(Tuple.Create(current.Name, "<could not calculate checksum>"));
+                }
+            }
+
+            references.Sort((x, y) => x.Item1.CompareTo(y.Item1));
+            foreach (var pair in references)
+            {
+                builder.AppendLine($"\t{pair.Item1} {pair.Item2}");
+            }
+        }
+
+        private string GetChecksum(string content)
+        {
+            var contentBytes = Encoding.UTF8.GetBytes(content);
+            var hashBytes = _hash.ComputeHash(contentBytes);
+            return HashBytesToString(hashBytes);
+        }
+
+        private static string HashBytesToString(byte[] hash)
+        {
+            var data = BitConverter.ToString(hash);
+            return data.Replace("-", "");
```

Comment:
Familiarity with git.  It feels more natural to me to see hashes in the raw, no-dash, form.  There  was no functional reason for this.